src/frontend/src/handler/create_source.rs

```diff
@@ -116,6 +117,11 @@ pub async fn handle_create_source(
     is_materialized: bool,
     stmt: CreateSourceStatement,
 ) -> Result<PgResponse> {
+    let has_pk = stmt.constraints.iter().any(|cons| match cons {
```

Comment:
The primary key can also be specified with column constraints. May check `row_id_column` for whether user pk exists.